salt/utils/event.py

```diff
@@ -401,9 +422,10 @@ def _get_event(self, wait, tag, match_func=None):
                 if socks.get(self.sub) != zmq.POLLIN:
                     continue
 
-                # Please do not use non-blocking mode here. Reliability is
-                # more important than pure speed on the event bus.
-                ret = self.get_event_block()
+                if no_block is True:
```

Comment:
The zmq docs say the timeout does the same as what select does, and from the select docs:

```
If timeout is given, it specifies the length of time in milliseconds which the system will wait for events before returning. If timeout is omitted, negative, or None, the call will block until there is an event for this poll object.
```

So, a timeout of 0 should be non-blocking. But as @s0undt3ch mentioned, even if its nonblocking, the while loop would make it spin. Actually in your case if there was no event you'd get an exception on `get_event_noblock` and the exception would go up to the caller.